### ISSUE TITLE
Wrong compiler warning reported for target-typed new under collection initilizer

### ISSUE THREAD

**Issue by controlflow** [dotnet/roslyn#60960]
**Version Used**: 

master branch

**Steps to Reproduce**:

Compile the following:
```c#
using System;
using System.Collections.Generic;

class A {
  [Obsolete("Message")]
  public A() { }
}

class B {
  // CS0618: 'A.A()' is obsolete: 'Message'
  List<A> x1 = new List<A> {new A()};

  // CS1062: The best overloaded Add method 'A.A()'
  // for the collection initializer element is obsolete. Message
  List<A> x2 = new List<A> {new()};
}
```

**Expected Behavior**:

Two CS0618 reported

**Actual Behavior**:

Two warnings reported:

> CS0618: 'A.A()' is obsolete: 'Message'
> CS1062: The best overloaded Add method 'A.A()' for the collection initializer element is obsolete. Message

Second one is misleading, since it mentions `List<T>.Add` method that is not actually obsolete

**Comment by controlflow** [comment]
We broke our codebase while massively refactoring code to use target-typed new feature, CS0618 was suppressed in a project with warnings-as-errors turned on, while CS1062 is not.

**Comment by jcouv** [comment]
FYI @alrz
